nx-dev/feature-ai/src/lib/feature-ai.tsx

```diff
@@ -0,0 +1,120 @@
+import { ReactNode, useState } from 'react';
+import { Button } from '@nx/nx-dev/ui-common';
+import { sendCustomEvent } from '@nx/nx-dev/feature-analytics';
+
+import { renderMarkdown } from '@nx/nx-dev/ui-markdoc';
+import { nxDevDataAccessAi } from '@nx/nx-dev/data-access-ai';
+
+export function FeatureAi() {
+  const [finalResult, setFinalResult] = useState<null | ReactNode>(null);
+  const [error, setError] = useState(null);
+  const [query, setSearchTerm] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const warning = `
+  {% callout type="warning" title="Always double check!" %}
+  This feature is still in Alpha.
+  The results may not be accurate, so please always double check with our documentation.
+{% /callout %}
+
+  `;
+
+  const handleSubmit = async () => {
+    setLoading(true);
+    let completeText = '';
+    let usage;
+    try {
+      const aiResponse = await nxDevDataAccessAi(query);
+      completeText = aiResponse.textResponse;
+      usage = aiResponse.usage;
+      setLoading(false);
+    } catch (error) {
+      setError(error as any);
+      setLoading(false);
+    }
+    sendCustomEvent('ai_query', 'ai', 'query', undefined, { query, ...usage });
```

Comment:
@bcabanes tagging you to see if GA looks good

Reply:
This is the right function, I let @isaacplmann check on the Google Analytics dashboard for how to properly set the dimensions.